DataFormats/Provenance/src/ProductNamePattern.cc

```diff
@@ -0,0 +1,106 @@
+#include <algorithm>
+#include <string>
+#include <string_view>
+#include <regex>
+#include <vector>
+
+#include <boost/algorithm/string/replace.hpp>
```

Comment:
The use of boost would need
```xml
<use name="boost_header"/>
```
in the `BuildFile.xml` (I believe `boost_header` is the correct tool)

Reply:
OK.

Yes, from a quick look, boost/algorithm is header-only.